R/register_handlers.R

```diff
@@ -0,0 +1,98 @@
+#' Register handlers for logging messages, warnings and errors
+#'
+#' @param namespace (`character(1)`) the logger namespace
+#' @param package (`character(1)`) the package name
+#'
+#' @return `NULL` invisibly. Called for its side effects.
+#'
+#' @details This function registers global handlers for messages, warnings and errors.
+#' The handlers will investigate the call stack and if it contains a function
+#' from the package, the message, warning or error will be logged into the respective
+#' namespace.
+#' The handlers are registered only once per package and type. Consecutive calls will be ignored.
+#'
+#' @note Registering handlers is disallowed within `tryCatch` blocks which is used throughout
+#' base R and other packages. Because of this, the `register_handlers` function is wrapped in a
+#' `try` block to avoid errors.
+#'
+#' @seealso [globalCallingHandlers()]
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' register_logger(namespace = "teal.logger")
+#' register_handlers("teal.logger")
+#' # see the outcome
+#' globalCallingHandlers()
+#' }
+register_handlers <- function(namespace, package = namespace) {
+  try(
+    {
+      register_handler_type(namespace = namespace, package = package, type = "message")
+      register_handler_type(namespace = namespace, package = package, type = "warning")
+      register_handler_type(namespace = namespace, package = package, type = "error")
+    },
+    silent = TRUE
+  )
+
+  invisible(NULL)
+}
+
+#' @keywords internal
+register_handler_type <- function(
+    namespace,
+    package,
+    type = c("error", "warning", "message")) {
+  match.arg(type)
+  if (!(is.character(namespace) && length(namespace) == 1)) {
+    stop("namespace argument must be a scalar character.")
+  }
+  if (!(namespace %in% logger::log_namespaces())) {
+    stop("namespace argument must be a pre-registered logger namespace.")
+  }
+  if (!(is.character(package) && length(package) == 1)) {
+    stop("package argument must be a scalar character.")
+  }
```

Comment:
```suggestion
  checkmate::assert_string(package)
```

I would not use `scalar` in terms of R. There are no scalars. Only vectors of length 1. With checkmate this is a short test, that has a meaningful message that detects the type and the lenght.